activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -170,6 +170,18 @@ def text_to_database_type(value)
             value
           end
         end
+
+        def database_type_to_text(value)
+          if value && cast_type.binary?
+            if cast_type.is_a?(ActiveRecord::Type::Serialized)
```

Comment:
When we want to serialize a binary column with `serialize :foo, JSON` and encrypt with `encrypts :foo`, the serialization steps should take place in the order:

```
Model            -> serializes            -> encrypts          -> db type (to db-encoded value)
(Ruby object)       JSON.dump to String      Encrypted String     Binary encoded DB value
```

And deserialization is the reverse

```
Database                  -> db type            -> encrypts           -> serializes
Binary encoded DB value      Binary String         Decrypted String      Ruby Object
```

Therefore I think it makes more sense to require that `encrypts :foo` is called first, then `serialize :foo`, as that will wrap the database type in the right order. With that we could get rid of the special case here.

@jorgemanrubia  - I've raised https://github.com/rails/rails/pull/52650 to show this simplification. Why is the [recommendation](https://guides.rubyonrails.org/active_record_encryption.html#supported-types) for `serialize`, then `encrypts`? I feel like I'm missing something here, is there a different case where applying them in the other order makes sense?